Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,540 @@
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+    IsScalarTower R C_c(α, β) C_c(α, β) where
+  smul_assoc r f g := by
+    ext
+    simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+instance {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+    SMulCommClass R C_c(α, β) C_c(α, β) where
+  smul_comm r f g := by
+    ext
+    simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section Star
+
+/-! ### Star structure
+
+It is possible to equip `C_c(α, β)` with a pointwise `star` operation whenever there is a continuous
+`star : β → β` for which `star (0 : β) = 0`. We don't have quite this weak a typeclass, but
+`StarAddMonoid` is close enough.
+
+The `StarAddMonoid` class on `C_c(α, β)` is inherited from their counterparts on `α →ᵇ β`.
+-/
+
+
+variable [TopologicalSpace β] [AddMonoid β] [StarAddMonoid β] [ContinuousStar β]
+
+instance : Star C_c(α, β) where
+  star f :=
+    { toFun := fun x => star (f x)
+      continuous_toFun := (map_continuous f).star
+      hasCompactSupport' := by
+        rw [HasCompactSupport, tsupport]
+        simp only
+        have support_star : (Function.support fun (x : α) => star (f x)) = Function.support f := by
+          ext x
+          simp only [Function.mem_support, ne_eq, star_eq_zero]
+        rw [support_star]
+        exact f.2 }
+
+@[simp]
+theorem coe_star (f : C_c(α, β)) : ⇑(star f) = star (⇑f) :=
+  rfl
+
+theorem star_apply (f : C_c(α, β)) (x : α) : (star f) x = star (f x) :=
+  rfl
+
+instance instStarAddMonoid [ContinuousAdd β] : StarAddMonoid C_c(α, β) where
+  star_involutive f := ext fun x => star_star (f x)
+  star_add f g := ext fun x => star_add (f x) (g x)
+
+end Star
+
+section StarModule
+
+variable {𝕜 : Type*} [Zero 𝕜] [Star 𝕜] [AddMonoid β] [StarAddMonoid β] [TopologicalSpace β]
+  [ContinuousStar β] [SMulWithZero 𝕜 β] [ContinuousConstSMul 𝕜 β] [StarModule 𝕜 β]
+
+instance : StarModule 𝕜 C_c(α, β) where
+  star_smul k f := ext fun x => star_smul k (f x)
+
+end StarModule
+
+section StarRing
+
+variable [NonUnitalSemiring β] [StarRing β] [TopologicalSpace β] [ContinuousStar β]
+  [TopologicalSemiring β]
+
+instance : StarRing C_c(α, β) :=
```

Comment:
Can you please add a `TrivialStar` instance under the assumption `[TrivialStar β]`?

Reply:
done.